### PR TITLE
Closes #172 Support for external OAuth tokens

### LINKED ISSUE #172: Support for external OAuth tokens
In order to support using [external OAuth tokens](https://docs.commercetools.com/http-api-authorization#requesting-an-access-token-using-an-external-oauth-server-beta), the SDK needs to be updated, so that the client application using the SDK can specify the token to be used in the `Authorization` headers.

Once a token is set by the client, it should be used in all subsequent requests to platform API. It should not be used for auth requests.

After a token is unset by the client, the SDK will either obtain new anonymous session token or plain token, depending on the configuration applied.

Open items:
- After the token is set by the client, should we allow `me/signup` and `me/login`?
@cneijenhuis would like to hear your thoughts on this.

### PR DESCRIPTION
**Pull request by nikola-mladenovic**
- [x] Add support for setting an external token by the client.
- [x] Auth manager tests.
- [x] Add `fatalError` to `loginCustomer` and `signUpCustomer` when using external OAuth.
- [x] Add external OAuth to readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,6 +156,10 @@ Commercetools.obtainAnonymousToken(usingSession: true, anonymousId: "some-custom
 ```
 When an anonymous sessions ends with a sign up or a login, carts and orders are migrated to the customer, and `CustomerSignInResult` is returned, providing access to both customer profile, and the currently active cart. For the login operation, you can define how to migrate line items from the currently active cart, by explicitly specifying one of two `AnonymousCartSignInMode` values: `.mergeWithExistingCustomerCart` or `.useAsNewActiveCustomerCart`.
 
+## External OAuth tokens
+
+Commercetools platform and the SDK provides the ability to use external OAuth tokens. In order to set a token from your app, use `Commercetools.externalToken` property. Once set, this token will be used for all platform requests from the SDK. In order to stop using external token, simply set this value to `nil`. To get more information on setting up and using external OAuth with commercetools platform, please refer to [this page](https://docs.commercetools.com/http-api-authorization#requesting-an-access-token-using-an-external-oauth-server-beta).
```
Comment (cneijenhuis) [inline README.md]: I'm assuming that the external token isn't valid "forever", and may even be revoked earlier under special circumstances.

Maybe you can add a sentence on how the SDK user should handle the case when the token isn't valid?

Comment (nikola-mladenovic) [reply README.md]: Yes, good point 👍

Comment (nikola-mladenovic) [reply README.md]: I've updated the readme - even though it is generic, the token refresh process is independent of the platform / SDK, so I'm not sure if there's more we can add. WDYT @cneijenhuis ?

Comment (cneijenhuis) [reply README.md]: That is perfect, as I said I didn't expect more than a sentence or two 😉

**Comment by nikola-mladenovic** [comment]
Great, thank you for the review 👍 🎉

**Pull request merged**